calculator.rb

```diff
@@ -0,0 +1,174 @@
+# Calculator
+# Platinum level
+# By Alice Rhomieux
+
+puts "Hello, welcome to the calculator."
+
+# List of supported operations, structured as a hash.
+# Each array is a family of terms that will be accepted for each operation.
+# The operator symbol is always the last element of the class so it can be selected with index [-1]
+# Oops, this is a global variable
+$supported_operations = {
+	addition: ["addition", "add", "plus", "+"],
+	subtraction: ["subtraction", "subtract", "minus", "sub", "-"],
+	multiplication: ["multiplication", "multiply", "times", "*"],
+	division: ["division", "divide", "div", "by", "/"],
+	exponent: ["exponent", "exp", "power of", "power", "**", "^"],
+	modulus: ["modulus", "modulo", "mod", "remainder", "%"]
+}
+
+# Check a number inputted by the user as a string and see if it has a decimal
+# If it has a decimal, convert it to a float, otherwise, convert it to an integer
+def check_num_format(input)
+	if input.include?(".")
+		data = input.delete(",").to_f
+	else
+ 		data = input.delete(",").to_i
+	end
+
+	return data
+end
+
+# Performe a calculation using an operation and two numbers
+# Return the output and the string form of the operator
+def calculate(operation, first_num, second_num)	
+
+	first_number_data = check_num_format(first_num)
+	second_number_data = check_num_format(second_num)
+	
+	if $supported_operations[:addition].include?(operation)
+		output = first_number_data + second_number_data
+		operator = $supported_operations[:addition][-1]
+
+	elsif $supported_operations[:subtraction].include?(operation)
+	 	output = first_number_data - second_number_data
+	 	operator = $supported_operations[:subtraction][-1]
+
+	elsif $supported_operations[:multiplication].include?(operation)
+	 	output = first_number_data * second_number_data
+	 	operator = $supported_operations[:multiplication][-1]
+
+	elsif $supported_operations[:division].include?(operation)
+	 	output = first_number_data.to_f / second_number_data.to_f
+	 	operator = $supported_operations[:division][-1]
+
+	elsif $supported_operations[:exponent].include?(operation)
+	 	output = first_number_data ** second_number_data
+	 	operator = $supported_operations[:exponent][-1]
+
+	elsif $supported_operations[:modulus].include?(operation)
+	 	output = first_number_data % second_number_data
+	 	operator = $supported_operations[:modulus][-1]
+	end
+
+	return output, operator
+end
+
+# Evaluate a statement contained in parentheses and return the value
+def evaluate_parens(statement_in_parens)
+	# strip the parens out of the input
+	statement = statement_in_parens[1...-1]
+	
+	# partition the statement using a regex to find the operation
+	# feed the elements of the array into the 2 numbers and operation
+	statement_first_num = statement.partition(/[\+\-\^\*\/\%]/)[0].strip
```

Comment:
Wow, that's awesome that you're using regex. :100: 

It looks like you're repeating `.partition` method. You can store that in a variable so you only have to call it once:
```
partitioned_statement = statement.partition(/[\+\-\^\*\/\%]/)
statement_first_num = partitioned_statement[0].strip
statement_operation = partitioned_statement[1].strip
statement_second_num = partitioned_statement[2].strip
```